### DISCUSSION TITLE
Why inject a null bean when i use CustomScope?

### DISCUSSION THREAD

**Discussion by Linindoo** [micronaut-projects/micronaut-core#6341] [Q&A] [answered]
i am use CustomScope to define my scope,if the beanDefine is not in my scope i will return a null bean.i can  use getBeansOfType method get all of bean,the collections has no null element.but it get NPE when if invoke the method of bean? i am confused why the null bean exist at the collections ?maybe it is a proxy but how can I judge that the proxy object of a bean is not null without NPE?

**Comment by graemerocher** [comment, marked as answer]
Perhaps related to https://github.com/micronaut-projects/micronaut-core/issues/6064

Report an issue with steps to reproduce and an example attached or supply info to the above issue
